Chore/bower to yarn

jimmyfortinx opened the pull request:
Migration of Bower to Yarn.

- grunt-wiredep removed
- a new dependencies.json file that contains all dependencies in it
- reuse grunt-injector to add dependencies inside the html

antleblanc commented on Makefile:
```diff
@@ -1,6 +1,7 @@
 NPM=npm
+YARN=yarn
 GRUNT=grunt
 BOWER=bower
```
Could be removed at the end.

jimmyfortinx commented:
@antleblanc I updated the `.editorconfig` and the `Makefile` to remove everything related to bower and I removed the `bower.json` and `.bowerrc` files. I had nothing to update inside the `.gitignore` since we are temporary creating a `bower_components` symlink under the `client` folder.
